client/common/common.c

```diff
@@ -516,6 +516,7 @@ menu_with_options(struct client *client)
         // bm_menu_grab_keyboard(menu, true);
         bm_menu_render(menu);
         bm_menu_set_filter(menu, NULL);
+        menu->dirty = true; // Ensure a redraw on the first call to bm_menu_render after the items are loaded
```

Comment:
The dirty is not considered public API. I think you should set this in this function instead https://github.com/Cloudef/bemenu/blob/84bccc02a0c851fd4933bae67b2f815e23e1e6e2/lib/menu.c#L175

Reply:
Done. Note that most `bm_menu_set_*` are missing the dirtiness check, which is what initially led me to try to just set the dirty flag and call it a day, but I agree that the ideal approach going forward should be to set it on all `bm_menu_set_*` functions that cause a change on the UI.